models/classes/task/AbstractStatementMigrationTask.php

```diff
@@ -0,0 +1,157 @@
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\tao\model\task;
+
+use common_exception_MissingParameter;
+use common_report_Report;
+use Iterator;
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\action\Action;
+use oat\oatbox\log\LoggerAwareTrait;
+use oat\tao\model\task\helper\PositionTracker;
+use oat\tao\model\task\helper\StatementLastIdRetriever;
+use oat\tao\model\task\helper\TaskIterator;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\tao\model\taskQueue\Task\CallbackTaskInterface;
+use oat\tao\model\taskQueue\Task\TaskAwareInterface;
+use oat\tao\model\taskQueue\Task\TaskAwareTrait;
+use Throwable;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+
+abstract class AbstractStatementMigrationTask implements Action, ServiceLocatorAwareInterface, TaskAwareInterface
+{
+    use ServiceLocatorAwareTrait;
+    use OntologyAwareTrait;
+    use TaskAwareTrait;
+    use LoggerAwareTrait;
+
+    /** @var int */
+    protected $affected;
+
+    /** @var int */
+    protected $pickSize;
+
+    /** @var common_report_Report */
+    protected $anomalies;
+
+    abstract protected function getTargetClasses(): array;
+
+    abstract protected function processUnit(array $unit): void;
+
+    public function __invoke($params)
+    {
+        if (count($params) < 4) {
+            throw new common_exception_MissingParameter();
+        }
+        $start = array_shift($params);
+        $chunkSize = array_shift($params);
+        $this->pickSize = $pickSize = array_shift($params);
+        $slideToTheStatementsEnd = (bool)array_shift($params);
+
+        $max = $this->getLastRowNumber();
+
+        $this->getServiceLocator()->get(PositionTracker::class)->keepCurrentPosition(static::class, $start);
+
+        $end = $start + $chunkSize;
+
+        if ($end >= $max) {
+            $end = $max;
+        }
+
+        $targetClasses = $this->getTargetClasses();
+        $this->initAnomaliesCollector();
+        $iterator = $this->getServiceLocator()->get(TaskIterator::class)->getIterator($targetClasses, $start, $end);
+
+        iterator_apply($iterator, [$this, 'applyProcessor'], [$iterator, $pickSize, $this->affected]);
```

Comment:
```suggestion
        iterator_apply($iterator, [$this, 'applyProcessor'], [$iterator]);
```

`applyProcessor` is only using one parameter. Can the others be removed?